toolkit/pallets/partner-chains-session/README.md

```diff
@@ -0,0 +1,180 @@
+# Partner Chains Session Pallet
+
+## Overview
+
+The Partner Chains Session pallet is a refined adaptation of Substrate's standard session pallet, customized to meet the requirements of partner chains while maintaining compatibility with core Substrate functionalities.
+
+This pallet serves as the cornerstone for managing validator sets and their associated session keys within the partner chain network. Unlike the original session pallet, it has been streamlined to eliminate unnecessary complexities while enhancing features that are critical for partner chains' consensus mechanisms.
+
+The Partner Chains Session pallet operates on the principle of session-based validator rotation, where:
+
+1. **Session Management**: A "session" represents a period during which a fixed set of validators is responsible for block production and finality. The pallet handles the orderly transition between sessions, ensuring consensus continuity.
+
+2. **Key Management**: Validators require various cryptographic keys for different consensus-related functions. The pallet provides a framework for validators to register and update their session keys securely.
+
+3. **Validator Rotation**: At the end of each session, the pallet can rotate to a new validator set, allowing the network to dynamically adjust its consensus participants based on various selection criteria.
+
+4. **Validator Disabling**: The pallet includes mechanisms to disable validators during a session if they're found to be misbehaving, enhancing network security and resilience.
+
+This pallet eliminates the complex queuing mechanism from the original session pallet, opting instead for a direct rotation model that aligns better with the epoch-based consensus used in partner chains. This design choice reduces state bloat and simplifies the mental model for developers working with the pallet.
+
+Furthermore, the pallet is designed to integrate seamlessly with other partner chain components, particularly the Session Validator Management pallet, creating a cohesive system for validator selection, rotation, and session management.
+
+## Purpose
+
+The Partner Chains Session pallet fulfills several critical functions within the partner chain ecosystem:
+
+- Manages the lifecycle of validator sessions, including session creation, rotation, and termination
+- Provides a secure mechanism for validators to associate their identity with necessary session keys
```

Comment:
I would say that the biggest benefit is that they are automatic. Otherwise validators would have to use session pallet `setKeys`.